scripts/release.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -e
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOT_DIR="${SCRIPT_DIR}/../"
+cd ${ROOT_DIR}
+
+print_usage() {
+  cat <<EOF
+Usage: $(basename $0) tag"
+
+Tag example: v1.2.3
+EOF
+}
+
+if [[ $# != 1 ]]
+then
+  print_usage
+  exit 1
+fi
+
+release_tag="$1"
+
+build_project() {
+  mkdir -p dist
+
+  # build the javaagent
+  ./gradlew -Prelease.useLastTag=true build final closeAndReleaseSonatypeStagingRepository -x test --no-daemon
+  mv agent/build/libs/splunk-otel-javaagent-*-all.jar dist/splunk-otel-javaagent-all.jar
+  mv agent/build/libs/splunk-otel-javaagent-*-all.jar.asc dist/splunk-otel-javaagent-all.jar.asc
+
+  # build the cloudfoundry buildpack
+  ./deployments/cloudfoundry/buildpack/build.sh
+  mv deployments/cloudfoundry/buildpack/splunk_otel_java_buildpack-linux.zip dist/
+  # TODO: sign the buildpack file?
+  # gpg --clear-sign file
+
+  # calculate checksums
+  shasum -a 256 dist/* > dist/checksums.txt
+  # TODO: sign the checksums file?
+}
+
+create_release() {
+  cat > release.json.tmp <<EOF
+{"tag_name":"${release_tag}","name":"Release ${release_tag}","draft":true}
+EOF
+  trap "rm release.json.tmp" EXIT
+
+  curl -s \
+    -X POST \
+    -H "Authorization: token $GITHUB_TOKEN" \
+    -H "Accept: application/vnd.github.v3+json" \
+    "https://api.github.com/repos/signalfx/splunk-otel-java/releases" \
+    --data @release.json.tmp | jq '.id'
+}
+
+upload_artifact() {
+  local gh_release_id="$1"
+  local artifact="$2"
+
+  curl -s \
+    -X POST \
+    -H "Authorization: token $GITHUB_TOKEN" \
+    -H "Content-Type: $(file -b --mime-type $artifact)" \
+    --data-binary @"$artifact" \
+    "https://uploads.github.com/repos/signalfx/splunk-otel-java/releases/${gh_release_id}/assets?name=$(basename $artifact)" > /dev/null
+}
+
+build_project
+gh_release_id=$(create_release)
+
+for f in dist/*
+do
+  upload_artifact "$gh_release_id" "$f"
+done
```

Comment:
are you sure you don't want to use existing `softprops/action-gh-release` action for that?

Reply:
As it is now we can't do signing in GH, so I can't...

Reply:
I don't see any mentioning of signing in that action description 🤔